src/SimpleThings/EntityAudit/AuditReader.php

```diff
@@ -819,4 +819,39 @@ public function getEntityHistory($className, $id)
 
         return $result;
     }
+    
+    public function findRevisionsByDateAndUsername($className, $id, \DateTime $dateFrom = null, \DateTime $dateTo = null, $username = null)
```

Comment:
Sorry, my suggestions might have been a bit confusing, sorry for that. Those were two separate comments on seperate parts of the code.

* array_filter was just meant to replace the foreach-loop
* the criteria object should only be a very simple class, a struct/value object with only public properties and no code - this was meant to keep the method signature small + it is easily extendable, just add another property to the criteria and implement the corresponding logic in the repository